scripts/analysis.py

```diff
@@ -56,188 +25,85 @@ def get_agent_ids(cursor, facility):
     cursor: cursor
         cursor for sqlite3
     facility: str
-        name of facility type
+        name of facility type as described in spec
 
     Returns
     -------
-
     sink_id: list
-        list of all the sink agentId values.
+        list of all the sink agentId values as string.
     """
+    agents = cursor.execute("SELECT * FROM agententry WHERE spec LIKE '%" +
+                            facility + "%' COLLATE NOCASE").fetchall()
 
-    cur = cursor
-    agent_id = []
-    agent = cur.execute("select * from agententry where spec like '%"
-                        + facility + "%'").fetchall()
+    return list(str(agent[1]) for agent in agents)
 
-    for ag in agent:
-        agent_id.append(ag[1])
-    return agent_id
 
-
-def get_waste_id(resource_list):
-
-    """ Gets waste id from a resource list
+def get_prototype_id(cursor, prototype):
+    """ Returns agentid of a prototype
 
     Parameters
-    ---------
-    resource_list: list
-        list fetched from the resource table.
+    ----------
+    cursor: sqlite cursor
+        sqlite cursor
+    prototype: str
+        name of prototype
 
     Returns
     -------
-    waste_id: list
-        list of qualId for waste
+    agent_id: list
+        list of agent_ids for prototype as string.
     """
+    ids = cursor.execute('SELECT agentid FROM agententry '
+                         'WHERE prototype = "' +
+                         str(prototype) + '" COLLATE NOCASE').fetchall()
 
-    wasteid = []
-
-    for res in resource_list:
-
-        wasteid.append(res[0])
+    return list(str(agent[0]) for agent in ids)
```

Comment:
Due to sqlite queries, getting a list of agentid strings turned out to be more useful than strings.
If this is not what you meant, I'm sorry but can you explain this to me please?